### ISSUE TITLE
Signature Verify warning in Syncfusion NuGet package

### ISSUE THREAD

**Issue by kalirajanganeshan** [NuGet/Home#11542]
### Impact

Other

### Describe the bug

We have signed and published the Syncfusion.Tools.Windows(v17.2.0.34) on 07/11/2019 at the time our signing certificate has validity till 2020. We are facing the below failure in below shared machine configuration while running the following NuGet signature verify command, but this issue was reproduced in some machine only. Can you please check and update solution for this issue?

**NuGet Signature Verify Command Details:**

nuget verify -Signatures {package directory}/syncfusion.tools.windows.17.2.0.34.nupkg} -Verbosity detailed

**nuget.exe version :** 5.11.0.10

**Machine Details**

Windows Server 2019 Standard
version: 1809
OS build Version: 17763.2183

**Issue details:**

EXEC : warning : NU3018: Package 'Syncfusion.Tools.Windows 17.2.0.34' from source 'C:\Users\{User Name}\.nuget\packages\': The author primary signature's signing certificate is not trusted by the trust provider.
The author primary signature validity period has expired.

WARNING: NU3018: The author primary signature found a chain building issue: The revocation function was unable to check revocation because the revocation server could not be reached. For more information, visit https://aka.ms/certificateRevocationMode.

### Repro Steps

1. Download [Syncfusion Tools windows package](https://www.nuget.org/packages/Syncfusion.Tools.Windows/17.2.0.34) in 17.2.0.34 version
2. Then run the following NuGet signature verify command in below machine configuration.

NuGet Signature Verify Command Details:

nuget verify -Signatures {package directory}/syncfusion.tools.windows.17.2.0.34.nupkg} -Verbosity detailed

nuget.exe version : 5.11.0.10

Machine Details

Windows Server 2019 Standard
version: 1809
OS build Version: 17763.2183

### Expected Behavior

- Signature expiry warnings wont be come while running NuGet signature verify command

**Comment by nkolev92** [comment]
Hey @kalirajanganeshan 

Do you see this issue with any other packages?

I'll try to get a machine to repro your scenario, but in the meantime you can hopefully help us clarify a few things about your scenario. 

Thanks!

**Comment by kalirajanganeshan** [comment]
Hey @nkolev92 ,

Please find the packages list which have this issue.

**Version**: 17.2.0.34

Syncfusion.Grid.Base
Syncfusion.Grid.Windows
SyncFusion.Shared.Base
SyncFusion.Shared.Windows
Sybcfusion.SpellChecker.Base
Syncfusion.Tool.BaseSyncfusion.Tool.Windows

Thanks,
Kalirajan Ganesan

**Comment by AbishakeD** [comment]
Hi ,

I’m also faced this issue. Could you please update on this?

Thanks,
Abishake

**Comment by kalirajanganeshan** [comment]
Hi,

Any update on this?

Thanks,
Kalirajan Ganesan

**Comment by kalirajanganeshan** [comment]
Hi @nkolev92 / @joelverhagen / @AArnott ,

Could you please update on this still we can't get rid of this issue?

Thanks,
Kalirajan Ganesan

**Comment by zivkan** [comment]
@kalirajanganeshan when I download syncfusion.grid.windows.19.4.0.53.nupkg from nuget.org, and run `dotnet nuget verify .\syncfusion.grid.windows.19.4.0.53.nupkg -v d`, I get the following output

<details>

<summary>dotnet nuget verify output</summary>

```text
Verifying Syncfusion.Grid.Windows.19.4.0.53
C:\Users\zivkan\Downloads\.\syncfusion.grid.windows.19.4.0.53.nupkg
Signature Hash Algorithm: SHA256

Signature type: Author
Verifying the author primary signature with certificate:
  Subject Name: CN=Syncfusion Inc, O=Syncfusion Inc, L=MORRISVILLE, S=North Carolina, C=US
  SHA1 hash: 648AA2E87F9B28053C4273493B20ACEFA70CC3BF
  SHA256 hash: 8F73976D6B12D2EC70700E3941081B686543079B0A96EEA496EB32175AFD03D5
  Issued by: CN=DigiCert SHA2 Assured ID Code Signing CA, OU=www.digicert.com, O=DigiCert Inc, C=US
  Valid from: 2020-08-10 5:00:00 PM to 2023-08-15 5:00:00 AM
trace:       Subject Name: CN=DigiCert SHA2 Assured ID Code Signing CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:       SHA1 hash: 92C1588E85AF2201CE7915E8538B492F605B80C6
trace:       SHA256 hash: 51044706BD237B91B89B781337E6D62656C69F0FCFFBE8E43741367948127862
trace:       Issued by: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:       Valid from: 2013-10-22 5:00:00 AM to 2028-10-22 5:00:00 AM
trace:             Subject Name: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:             SHA1 hash: 0563B8630D62D75ABBC8AB1E4BDFB5A899B24D43
trace:             SHA256 hash: 3E9099B5015E8F486C00BCEA9D111EE721FABA355A89BCF1DF69561E3DC6325C
trace:             Issued by: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:             Valid from: 2006-11-09 4:00:00 PM to 2031-11-09 4:00:00 PM
Timestamp: 2022-02-21 12:46:59 PM
Verifying author primary signature's timestamp with timestamping service certificate:
  Subject Name: CN=DigiCert Timestamp 2021, O="DigiCert, Inc.", C=US
  SHA1 hash: E1D782A8E191BEEF6BCA1691B5AAB494A6249BF3
  SHA256 hash: B3109006BC0AD98307915729E04403415C83E3292B614F26964C8D3571ECF5A9
  Issued by: CN=DigiCert SHA2 Assured ID Timestamping CA, OU=www.digicert.com, O=DigiCert Inc, C=US
  Valid from: 2020-12-31 4:00:00 PM to 2031-01-05 4:00:00 PM
trace:       Subject Name: CN=DigiCert SHA2 Assured ID Timestamping CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:       SHA1 hash: 3BA63A6E4841355772DEBEF9CDCF4D5AF353A297
trace:       SHA256 hash: CA8D0F4736454AECBEC5DEEC80998C9EBF41D06C728F3C76CCA24151BC62D463
trace:       Issued by: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:       Valid from: 2016-01-07 4:00:00 AM to 2031-01-07 4:00:00 AM
trace:             Subject Name: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:             SHA1 hash: 0563B8630D62D75ABBC8AB1E4BDFB5A899B24D43
trace:             SHA256 hash: 3E9099B5015E8F486C00BCEA9D111EE721FABA355A89BCF1DF69561E3DC6325C
trace:             Issued by: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:             Valid from: 2006-11-09 4:00:00 PM to 2031-11-09 4:00:00 PM

Signature type: Repository
Service index: https://api.nuget.org/v3/index.json
Owners: SyncfusionInc
Verifying the repository countersignature with certificate:
  Subject Name: CN=NuGet.org Repository by Microsoft, O=NuGet.org Repository by Microsoft, L=Redmond, S=Washington, C=US
  SHA1 hash: 68FC37CCC293766A607DBC26A26DF371B0249328
  SHA256 hash: 5A2901D6ADA3D18260B9C6DFE2133C95D74B9EEF6AE0E5DC334C8454D1477DF4
  Issued by: CN=DigiCert SHA2 Assured ID Code Signing CA, OU=www.digicert.com, O=DigiCert Inc, C=US
  Valid from: 2021-02-15 4:00:00 PM to 2024-05-15 4:59:59 PM
trace:       Subject Name: CN=DigiCert SHA2 Assured ID Code Signing CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:       SHA1 hash: 92C1588E85AF2201CE7915E8538B492F605B80C6
trace:       SHA256 hash: 51044706BD237B91B89B781337E6D62656C69F0FCFFBE8E43741367948127862
trace:       Issued by: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:       Valid from: 2013-10-22 5:00:00 AM to 2028-10-22 5:00:00 AM
trace:             Subject Name: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:             SHA1 hash: 0563B8630D62D75ABBC8AB1E4BDFB5A899B24D43
trace:             SHA256 hash: 3E9099B5015E8F486C00BCEA9D111EE721FABA355A89BCF1DF69561E3DC6325C
trace:             Issued by: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:             Valid from: 2006-11-09 4:00:00 PM to 2031-11-09 4:00:00 PM
Timestamp: 2022-02-21 4:03:27 PM
Verifying repository countersignature's timestamp with timestamping service certificate:
  Subject Name: CN=Symantec SHA256 TimeStamping Signer - G3, OU=Symantec Trust Network, O=Symantec Corporation, C=US
  SHA1 hash: A9A4121063D71D48E8529A4681DE803E3E7954B0
  SHA256 hash: C474CE76007D02394E0DA5E4DE7C14C680F9E282013CFEF653EF5DB71FDF61F8
  Issued by: CN=Symantec SHA256 TimeStamping CA, OU=Symantec Trust Network, O=Symantec Corporation, C=US
  Valid from: 2017-12-22 4:00:00 PM to 2029-03-22 4:59:59 PM
trace:       Subject Name: CN=Symantec SHA256 TimeStamping CA, OU=Symantec Trust Network, O=Symantec Corporation, C=US
trace:       SHA1 hash: 6FC9EDB5E00AB64151C1CDFCAC74AD2C7B7E3BE4
trace:       SHA256 hash: F3516DDCC8AFC808788BD8B0E840BDA2B5E23C6244252CA3000BB6C87170402A
trace:       Issued by: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
trace:       Valid from: 2016-01-11 4:00:00 PM to 2031-01-11 3:59:59 PM
trace:             Subject Name: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
trace:             SHA1 hash: 3679CA35668772304D30A5FB873B0FA77BB70D54
trace:             SHA256 hash: 2399561127A57125DE8CEFEA610DDF2FA078B5C8067F4E828290BFB860E84B3C
trace:             Issued by: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
trace:             Valid from: 2008-04-01 5:00:00 PM to 2037-12-01 3:59:59 PM

Successfully verified package 'Syncfusion.Grid.Windows.19.4.0.53'.
```

</details>

So, it appears that the author certificate's root certificate is:

```text
trace:             Subject Name: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:             SHA1 hash: 0563B8630D62D75ABBC8AB1E4BDFB5A899B24D43
trace:             SHA256 hash: 3E9099B5015E8F486C00BCEA9D111EE721FABA355A89BCF1DF69561E3DC6325C
trace:             Issued by: CN=DigiCert Assured ID Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US
trace:             Valid from: 2006-11-09 4:00:00 PM to 2031-11-09 4:00:00 PM
```

and the repo signature's root certificate is:

```text
trace:             Subject Name: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
trace:             SHA1 hash: 3679CA35668772304D30A5FB873B0FA77BB70D54
trace:             SHA256 hash: 2399561127A57125DE8CEFEA610DDF2FA078B5C8067F4E828290BFB860E84B3C
trace:             Issued by: CN=VeriSign Universal Root Certification Authority, OU="(c) 2008 VeriSign, Inc. - For authorized use only", OU=VeriSign Trust Network, O="VeriSign, Inc.", C=US
trace:             Valid from: 2008-04-01 5:00:00 PM to 2037-12-01 3:59:59 PM
```

Make sure these certificates are in your trusted CA certificates list. If your computer is running Windows and on an Active Directory domain, check that the domain admins haven't used Group Policy to remove trusted CA certs that Windows Update would normally install.


I now see that your were talking about a different package, that used a different certificate.  You can repeat the steps, check verbose output for the `verify` command, check the certificate chain, and ensure that the root certificate is trusted on the machines where it's working for you, and not trusted on the machines where it's not working for you.

**Comment by nkolev92** [comment]
Thank you reporting this issue. At this point we do not have enough information to continue investigating the issue. 
Please feel free to comment or open a new issue if needed.

**Comment by AbishakeD** [comment]
Hi Team,

We checked our NuGet signature certificates in trusted CA certificate list from our machine. Those certificates are trusted on that machine but still nuget signature verification command warning issue are persists.

Could you please check and update on this?


Thanks,
Abishake

**Comment by kalirajanganeshan** [comment]
Hi Team,

Any update on this?

Thanks,
Kalirajan Ganesan

**Comment by kalirajanganeshan** [comment]
Hi Team,

Can you please update on this?

Thanks,
Kalirajan Ganesan

**Comment by joelverhagen** [comment]
Hey @kalirajanganeshan and @AbishakeD, I think there's a mismatch in expectations on closed GitHub issues. On issues in NuGet repositories, and in my experience on GitHub in general, closed issues are a difficult place to conduct a conversation because the state of the issue suggests the problem is no longer being actively attended to by the maintainers. Oftentimes management bots, notification settings, triage processes, etc all filter out activity on closed issues.

For this particular issue, I recommend investigating the problem across several machine configurations (e.g. home machine, work machine, Azure VM) or providing a minimal repro which we can reliably test on our end (e.g. a set of steps including a clean PC environment we can set up on our end, like a Microsoft-published Azure VM image on an Azure VM).

@zivkan was unable to reproduce the problem and, as @nkolev92 mentioned, we don't have enough information to proceed.

I suggest one of the two courses of action:

**Option 1**: deeply investigate the difference in trusted root CAs on your machine vs. another machine you have access to which does not reproduce the issue (e.g. a home PC). My gut feeling is that this could be related to your trust of the _timestamp_ CA on the repository signature, which is Symtanec, as shown in @zivkan's sample output above. This comparison may allow you to identify a missing root CA or some other difference in environment.

**Option 2**: reproduce the issue in a clean environment which, say, anyone with an Azure subscription can perform, and provide us the steps in a **new GitHub issue**. This will allow us to reliably get an environment that matches the scenario so we can identify a root cause. Feel free to link to this issue for reference.

In summary, we do not have enough information to reproduce the problem so you can consider investigating trust settings more deeply or providing us with a reliable repro. Once you know more or have a repro, please feel free to open a new issue and the team will respond. I hope this helps!
